src/router/index.ts

```diff
@@ -1,17 +1,101 @@
 import h from "hyperscript"
 import cx from "classnames"
+import { ROUTES } from "~router/routes"
+import { AppEvent, dispatch } from "~pages/app/events"
+import { rootElement } from "~index"
 
 export type LinkProps = {
   path: string
   title: string
   className?: string
+  children: Node | Node[]
 }
-export function Link({ path, title, className }: LinkProps) {
+export function Link({ path, title, children, className }: LinkProps) {
   function handlClick(e: MouseEvent) {
     e.preventDefault()
-    history.pushState(null, title, path)
+    navigateTo(path, title)
   }
 
-  return h("a", { className: cx("link", className), href: path, ["onclick"]: handlClick, title })
+  return h(
+    "a",
+    { className: cx("link", className), href: path, ["onclick"]: handlClick, title },
+    children
+  )
+}
+
+function isRelativePath(path: string) {
+  console.log(path)
+  return path[0] !== "/"
+}
+
+export function navigateTo(path: string, title: string) {
+  const cleansedPath = cleanPath(path)
+  const isRelative = isRelativePath(cleansedPath)
+  const redirectTo = isRelative ? `${getPathFromWindowUrl()}/${cleansedPath}` : cleansedPath
+  console.log({ isRelative, redirectTo })
+  pushState(redirectTo, title)
+  updateRoute(redirectTo)
+}
+
+export function pushState(path: string, title: string) {
+  history.pushState(null, title, path)
+}
+
+function cleanPath(path: string) {
+  if (path === "/") {
+    return path
+  }
+  return path.replace(/\/$/, "")
+}
+
+function getPathFromWindowUrl() {
+  return cleanPath(window.location.pathname)
+}
+
+function matchPath(path: string) {
```

Comment:
goes through known routes defined in `ROUTES` configuration, and tries to match the path given it with an existing route